tests/test_prepare_annotations.py

```diff
@@ -100,3 +103,146 @@ def __prepare_pydantic_annotations__(
 
     a = TypeAdapter(Annotated[MyType, Lt(420)])
     assert a.core_schema == de.IsPartialDict(core_schema.int_schema(gt=123, lt=420))
+
+
+roman_numbers = {
+    'I': 1,
+    'II': 2,
+    'III': 3,
+    'IV': 4,
+    'V': 5,
+    'VI': 6,
+    'VII': 7,
+    'VIII': 8,
+    'IX': 9,
+    'X': 10,
+}
+roman_numbers_lookup = {v: k for k, v in roman_numbers.items()}
+
+
+class Roman:
```

Comment:
this example is added to demonstrate use of `config` argument to `__prepare_pydantic_annotations__`.

It's quite understandable I think although a bit long, I guess it could make its way into the docs to demonsrate use of `__prepare_pydantic_annotations__`.